File: app/src/main/java/com/olehmesh/randomusers_task/adapters/UsersAdapter.java
package com.olehmesh.randomusers_task.adapters;

import android.content.Context;
import android.content.Intent;
import androidx.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;
import com.bumptech.glide.request.target.Target;
import com.olehmesh.randomusers_task.R;
import com.olehmesh.randomusers_task.models.Result;
import com.olehmesh.randomusers_task.views.DetailActivity;

import java.util.List;

public class UsersAdapter extends RecyclerView.Adapter<ItemViewHolder> {

    Context context;
    List<Result> mlist;
    Intent intent;

    public UsersAdapter(Context context, List<Result> list) {
        this.context = context;
        this.mlist = list;

    }

    @NonNull
    @Override
    public ItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item, parent, false);
        return new ItemViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull final ItemViewHolder itemViewHolder, int i) {
        itemViewHolder.list = mlist.get(i);

        itemViewHolder.tvName.setText(mlist.get(i).getName().getFirst());
        itemViewHolder.tvCity.setText(mlist.get(i).getLocation().getCity());

        Glide.with(itemViewHolder.itemView.getContext())
                .asBitmap()
                .load(mlist.get(i).getPicture().getLarge())
                .apply(new RequestOptions()
                        .override(Target.SIZE_ORIGINAL).encodeQuality(100))
                .placeholder(R.drawable.placeholder)
                .error(R.drawable.placeholder)
                .into(itemViewHolder.ivAvatar);

        itemViewHolder.setOnItemClickListener((view, position, isLongClick) -> {

            intent = new Intent(view.getContext(), DetailActivity.class);
            intent.putExtra(String.valueOf(R.string.name), itemViewHolder.tvName.getText());
            intent.putExtra(String.valueOf(R.string.city), itemViewHolder.tvCity.getText());


            intent.putExtra(String.valueOf(R.string.image), mlist.get(position).getPicture().getLarge());

            intent.putExtra(String.valueOf(R.string.email), mlist.get(position).getEmail());
            intent.putExtra(String.valueOf(R.string.phone), mlist.get(position).getPhone());

            view.getContext().startActivity(intent);

        });

    }

    @Override
    public int getItemCount() {
        if (mlist == null) {
            return 0;
        }
        return mlist.size();
    }

    public void setData(List<Result> mlist) {
        this.mlist = mlist;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/olehmesh/randomusers_task/adapters/UsersAdapter.kt
package com.olehmesh.randomusers_task.adapters

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.NavController
import androidx.navigation.NavOptions
import androidx.navigation.Navigation
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.bumptech.glide.request.target.Target
import com.olehmesh.randomusers_task.Constants
import com.olehmesh.randomusers_task.R
import com.olehmesh.randomusers_task.databinding.MainListItemBinding
import com.olehmesh.randomusers_task.models.Result
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.main_list_item.*

class UsersAdapter : RecyclerView.Adapter<UsersAdapter.ItemViewHolder>() {

    private lateinit var navController: NavController
    private lateinit var mList: List<Result>

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {

        val v = LayoutInflater.from(parent.context)

        val binding = MainListItemBinding.inflate(v, parent, false)

        return ItemViewHolder(binding)
    }

    override fun onBindViewHolder(itemViewHolder: ItemViewHolder, i: Int) {

        itemViewHolder.bind(mList[i])
        itemViewHolder.itemView.setOnClickListener {

            navController = Navigation.findNavController(itemViewHolder.itemView)
            val builder = NavOptions.Builder()
            val navOptions = builder
                .setEnterAnim(R.anim.slide_in_right)
                .setExitAnim(R.anim.slide_out_right)
                .setPopEnterAnim(R.anim.slide_in_left)
                .setPopExitAnim(R.anim.slide_out_left)
                .build()

            val bundle = Bundle()

            bundle.putString(Constants.NAME, mList[i].name!!.first)
            bundle.putString(Constants.CITY, mList[i].location!!.city)
            bundle.putString(Constants.IMAGE, mList[i].picture!!.large)
            bundle.putString(Constants.EMAIL, mList[i].email)
            bundle.putString(Constants.PHONE, mList[i].phone)

            navController.navigate(R.id.fragment_detail, bundle, navOptions)

        }

    }

    override fun getItemCount(): Int {
        return mList.count()
    }

    fun setData(mList: List<Result>) {
        this.mList = mList

        notifyDataSetChanged()
    }

    inner class ItemViewHolder(private val binding: MainListItemBinding) :
        RecyclerView.ViewHolder(binding.root), LayoutContainer {

        override val containerView: View?
            get() = itemView

        fun bind(item: Result) {

            binding.result = item

            Glide.with(itemView.context)
                .load(item.picture!!.large)
                .apply(RequestOptions().override(Target.SIZE_ORIGINAL).encodeQuality(100))
                .placeholder(R.drawable.placeholder)
                .error(R.drawable.placeholder)
                .into(ivAvatar)

        }

    }
}